enhancements/network/admin-network-policy.md

```diff
@@ -0,0 +1,322 @@
+* Provide support for admins to define `Pass` action rules in ANP API that will enable
+interaction between ANP & NP OR ANP & BANP
+
+### Non-Goals
+
+* v1alpha1 version of the API only supports east-west traffic scenarios for ANP and BANP. Egress
+and ingress traffic policies at cluster-scoped level will be [iterated upon in the future
+versions of the API](https://github.com/kubernetes-sigs/network-policy-api/pull/86)
+* Two fields in the API will not be implemented in the first iteration because
+they are subject to change in the future versions of the API. Hence we will hold off
+until these parts are stable before implementing them downstream:
+  * [`sameLabels`](https://github.com/kubernetes-sigs/network-policy-api/blob/7321d8509d9bf425c9c5b0338db3c0fb2816b4c5/apis/v1alpha1/shared_types.go#L134)
+  * [`notSameLabels`](https://github.com/kubernetes-sigs/network-policy-api/blob/7321d8509d9bf425c9c5b0338db3c0fb2816b4c5/apis/v1alpha1/shared_types.go#L143)
+* ANP & BANP Policy logging (will be iterated upon in the future revisions)
+* ANP & BANP Scale testing (will be iterated upon in the future revisions)
+* Supporting ANP&BANP on Secondary Networks. Today Multi-Network Policies are
+maintained by https://github.com/openshift/multus-networkpolicy and any future
+support for cluster-scoped will have to come from those working groups to maintain parity.
+
+
+The implementation will be an iterative process based on customer feedback and performance evaluation. Goal is to get a tech preview version of the current v1alpha1 API in OCP 4.14 which will be improved upon.
+
+## Proposal
+
+A new level-driven controller will be added in OVN-Kubernetes which will
+watch for the `AdminNetworkPolicy`, `BaselineAdminNetworkPolicy`, `Namespaces`
+and `Pods` objects and create the:
+
+* port groups for the subject pods selected in the policies - we will try to
+investigate if shared port groups can be used here whereever possible
+* address-sets for the peers defined in the policies - we will try and reuse
+the address sets that are created for namespaces today and share them wherever possible
+* acls for the ingress and egress policy rules that will connect these port-groups
+and address-sets together into meaningful allow, deny or pass rules
+
+for:
+
+* ANP add/update/delete
+* BANP add/update/delete
+* pod add/update/delete (if it matches the policies; NOTE: updates are handled
+only if pod labels, podIPs or podState (completed) changed)
+* namespace add/update/delete (if it matches the policies; NOTE: updates are
+handled only for label changes)
+
+This controller will live in the `go-controller/pkg/ovn/controller` path
+of OVNKubernetes repo under its own package `admin_network_policy`. Since these
+are new APIs, we cannot reuse the [caches](https://github.com/ovn-org/ovn-kubernetes/blob/68295ec197b8eadd61ff872e293676fa1a68179d/go-controller/pkg/ovn/base_network_controller_policy.go#L123) that are present in network policy code.
```

Comment:
```suggestion
are new APIs, we cannot reuse the [caches](https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/base_network_controller_policy.go#L123) that are present in network policy code.
```

Reply:
Wait is this the right link?

Reply:
yeah I marked the perma link to the code line, if I use `master` it will change when the code moves/changes
https://github.com/ovn-org/ovn-kubernetes/blob/68295ec197b8eadd61ff872e293676fa1a68179d/go-controller/pkg/ovn/base_network_controller_policy.go#L123
should work